tests/Unit/Services/Events/ClientTest.php

```diff
@@ -119,6 +126,36 @@ public function testDirectCatalogActions()
                 $this->equalTo('post'),
                 $this->equalTo('categories'),
                 ['query' => ['service' => 'test', Base::KEY_TYPE => Base::SYNC], 'json' => '{"categories":[[]]}']
+            ],
+            [
```

Comment:
this has gotten a bit too lengthy, it's best if you separate this testMethod into Categories and Products separately